### PR TITLE
Switching hugger uses radial + keybind for it

### PR DESCRIPTION
**Pull request by BraveMole**
## About The Pull Request

Title

## Why It's Good For The Game

More convenient

## Changelog
:cl:
qol: Switching hugger now has 2 keybinds, one gives a radial menu to choose the hugger and the other switches in old fashioned way.
/:cl:

### CODE REVIEW

**Comment by Al-1ce** [comment]
You can cook huggies still, right? If so Carrier may become god tier.

**Review by SplinterGP — APPROVED** [review]

File: code/modules/mob/living/carbon/xenomorph/castes/carrier/abilities_carrier.dm [hunk 1]
```diff
@@ -1,11 +1,26 @@
...
+		))
+
+//List of huggie images
+GLOBAL_LIST_INIT(hugger_images_list,  list(
+		LARVAL_HUGGER = image('icons/mob/actions.dmi', icon_state = LARVAL_HUGGER),
+		CLAWED_HUGGER = image('icons/mob/actions.dmi', icon_state = CLAWED_HUGGER),
+		NEURO_HUGGER = image('icons/mob/actions.dmi', icon_state = NEURO_HUGGER ),
+		ACID_HUGGER  = image('icons/mob/actions.dmi', icon_state = ACID_HUGGER),
+		RESIN_HUGGER = image('icons/mob/actions.dmi', icon_state = RESIN_HUGGER),
+		))
```
Comment (SplinterGP) [inline code/modules/mob/living/carbon/xenomorph/castes/carrier/abilities_carrier.dm:23]: try doing this using less assoc possible, as assoc lists are heavier than normal lists or vars

**Review by SplinterGP — CHANGES_REQUESTED** [review]

File: code/__DEFINES/dcs/signals.dm [hunk 2]
```diff
@@ -523,6 +523,8 @@
 #define COMSIG_XENOABILITY_RETRIEVE_EGG "xenoability_retrieve_egg"
 #define COMSIG_XENOABILITY_PLACE_TRAP "xenoability_place_trap"
 #define COMSIG_XENOABILITY_SPAWN_HUGGER "xenoability_spawn_hugger"
+#define COMSIG_XENOABILITY_SWITCH_HUGGER "xenoability_switch_hugger"
+#define COMSIG_XENOABILITY_CHOOSE_HUGGER "xenoability_chooose_hugger"
```
Comment (BraveMole) [inline code/__DEFINES/dcs/signals.dm:527]:
```suggestion
#define COMSIG_XENOABILITY_CHOOSE_HUGGER "xenoability_choose_hugger"
```

File: code/modules/mob/living/carbon/xenomorph/castes/carrier/abilities_carrier.dm [hunk 3]
```diff
@@ -1,11 +1,26 @@
...
 //List of huggie types
-GLOBAL_LIST_INIT(hugger_type_list, typecacheof(list(
+GLOBAL_LIST_INIT(hugger_type_list, list(
 		/obj/item/clothing/mask/facehugger/larval,
 		/obj/item/clothing/mask/facehugger/slash,
 		/obj/item/clothing/mask/facehugger/neuro,
 		/obj/item/clothing/mask/facehugger/acid,
 		/obj/item/clothing/mask/facehugger/resin,
-		)))
+		))
```
Comment (SplinterGP) [inline code/modules/mob/living/carbon/xenomorph/castes/carrier/abilities_carrier.dm:14]: unneeded list can use typesof?

File: code/modules/mob/living/carbon/xenomorph/castes/carrier/abilities_carrier.dm [hunk 4]
```diff
@@ -247,3 +263,16 @@ GLOBAL_LIST_INIT(hugger_type_list, typecacheof(list(
 	to_chat(X, "<span class='notice'>We will now spawn <b>[initial(A.name)]\s</b> when using the Spawn Hugger ability.</span>")
 	update_button_icon()
 	return succeed_activate()
+
+/datum/action/xeno_action/choose_hugger_type/action_activate()
+	var/hugger_choice = show_radial_menu(owner, owner, GLOB.hugger_images_list, radius = 48)
+	if(!hugger_choice)
+		return
+	var/mob/living/carbon/xenomorph/X = owner
+	for(var/obj/item/clothing/mask/facehugger/hugger_type AS in GLOB.hugger_type_list)
```
Comment (SplinterGP) [inline code/modules/mob/living/carbon/xenomorph/castes/carrier/abilities_carrier.dm:272]:
```suggestion
	for(var/obj/item/clothing/mask/facehugger/hugger_type AS in typesof(/obj/item/clothing/mask/facehugger))
```

Comment (SplinterGP) [reply code/modules/mob/living/carbon/xenomorph/castes/carrier/abilities_carrier.dm:272]: or subtypes of i dont know if we have that one

**Review by SplinterGP — CHANGES_REQUESTED** [review]

**Pull request merged**